agent/app/agent.go

```diff
@@ -579,3 +588,24 @@ func (agent *ecsAgent) startACSSession(
 	seelog.Critical("ACS Session handler should never exit")
 	return exitcodes.ExitError
 }
+
+// validateRequiredDockerVersion validates docker version to be minimum 1.9.0
+func (agent *ecsAgent) validateRequiredDockerVersion(selector string) error {
+	dockerVersion, err := agent.dockerClient.Version(agent.ctx, dockerclient.VersionTimeout)
+	if err != nil {
```

Comment:
Should we retry in the case of a timeout and not exit with a terminal code?

Reply:
it will exit with terminal code, since timeout will result in error.

Reply:
I was wondering if timeout was a transient issue, it might make sense for init to try and restart agent again to find the docker version. So instead of returning a terminal code, we could exit with a non terminal one. What do you think?

Reply:
if there's a timeout, we could exit with a non terminal error code imo